lib/storage/environment.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/gravity/lib/constants"
+	"github.com/gravitational/gravity/lib/defaults"
+
+	teleservices "github.com/gravitational/teleport/lib/services"
+	teleutils "github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+)
+
+// Environment defines the environment variables resource.
+// It allows to override environment variables on each node in the cluster.
+// There is only a single instance of the resource in a cluster
```

Comment:
So I've honestly always found this a bit weird about how gravity emulates kubernetes resource type behaviour. We defined the object like a kubernetes object, with a name+namespace, but then the management of that is just a flat configuration that ignores those fields.